lib/path.js

```diff
@@ -1136,6 +1129,22 @@ const win32 = {
     else if (isAbsolute)
       ret.dir = path.slice(0, rootEnd);
 
+    Object.defineProperty(ret, 'base', {
+      enumerable: true,
+      configurable: false,
+      get() { return this.name + this.ext; },
+      set(value) {
+        if (value.startsWith('.') || !value.includes('.')) {
```

Comment:
Why do we need both of these? Can't we combine both using one call to `value.indexOf()`?:

```js
var dotIdx = value.indexOf('.');
if (dotIdx === 0 || dotIdx === -1) {
  // ...
```